### PR TITLE
Restore the installation repositories after going back

### PR DESCRIPTION
**Pull request by lslezak**
## Fix

- Fixed restoring the initial installation repositories
  - The old code just kept the first repository, that removed all additional repositories (DUD, custom addons)
  - The migration rollback reloads all repositories, that first repository might not be the initial one
  - Now really save the repositories and restore them
- Tested manually, now you can go back and forth in the upgrade workflow, upgrade the registration to SLE15, downgrade back to SLE12 and try again (with a different or even the same partition)
- Sorry for the coverage drop, adding the test for the client would require some refactoring... :confused:
  I'm focused on killing bugs in the RC phase.
- 4.0.11

## Extra Cleanup

- Cleanup, removed old unused files (I'll update `yast2-packager` to not use the dropped scripts)
- Remove the historic unsupported file systems (+ ReiserFS which has been dropped recently)
- Make sure we run the tests in English locale

### CODE REVIEW

File: src/bin/remove_junk [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-
-if test -d $1/var/lib/YaST2 ; then
-  # place here files from dropped modules
-fi
-
-# remove old you stuff which can be quite big
-if test -d $1/var/lib/YaST2/you ; then
-    /bin/rm -f $1/var/lib/YaST2/you/quickcheck
-    /bin/rm -fr $1/var/lib/YaST2/you/{files,installed,mnt,scripts}
-fi
```
Comment (jreidinger) [inline src/bin/remove_junk]: we do not need this mess if we properly use ghost files in past :)

File: src/modules/RootPart.rb [hunk 2]
```diff
@@ -1127,11 +1127,8 @@ def MountFSTab(fstab, message)
         "ext3",
         "ext4",
         "btrfs",
-        "minix",
-        "reiserfs",
```
Comment (jreidinger) [inline src/modules/RootPart.rb]: just are you sure that we without this can report error that reiserfs is not supported?

Comment (lslezak) [reply src/modules/RootPart.rb]: I see, let me check that...

Comment (lslezak) [reply src/modules/RootPart.rb]: I turned out that I still have a SLES10 VM with ReiserFS, so I could test that.

And yes it works:

sles10_upgrade_reiserfs

**Pull request merged**